apps/managedidentity/managedidentity.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/*
+Package managedidentity provides a client for retrieval of Managed Identity applications.
+The Managed Identity Client is used to acquire a token for managed identity assigned to
+an azure resource such as Azure function, app service, virtual machine, etc. to acquire a token
+without using credentials.
+*/
+package managedidentity
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/internal/base"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/internal/oauth/ops"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/internal/oauth/ops/authority"
+)
+
+const (
+	// AzureArc represents the source to acquire token for managed identity is Azure Arc.
+	AzureArc = 0
```

Comment:
Do the numbers matter? In languages like C# and Java, 0 is reserved for a value indicating "none" or "unknown" or smth.

Reply:
The numbers just ensure that the corresponding value always matches that number,
previously we had something like this:

```
const (
    Planet = iota // 0
    Star               // 1
    Comet           // 2
    Galaxy           // 3
)
```

however Charles notified us that if you do it this way, and add another value to this const list at a later stage, it increments the numbers:

```
const (
    Planet = iota // 0
    Star               // 1
    Earth             // 2
    Comet           // 3
    Galaxy           // 4
)
```

So if we ever relied on the value of 2 for Comet previously, it will now be different and relate to Earth, so putting the numerical value here just ensures that those values stay the same

Reply:
There's another subtlety here--the constant having value 0 becomes the default because the underlying type is `int` (more details in https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/498#discussion_r1725666156). Whether this matters depends on how the constant is used. I only point it out for consideration in case you want a different default value or to identify when a field hasn't been assigned.

Reply:
I see, in that case, should we have 0 be None like what was discussed before, or have IMDS or AzureArc be 0, which would be the default.
I think since we usually default to imds, we could make DefaultToIMDS = 0 instead. Make sense @bgavrilMS